### DISCUSSION TITLE
Update to 1.11.28 (quickly!)

### DISCUSSION THREAD

**Discussion by ywarnier** [chamilo/chamilo-lms#5884] [General]
Hi,

1.11.28 has just been released during the night. Together with a few new features, Chamilo 1.11.28 is *most of all* a security update version that adds fixes to a record number of potential vulnerabilities, one of which (at least) being of critical impact.
We urge every Chamilo admin out there to update to 1.11.28 as soon as possible.
You can find more details about the security patches (from #132 to #180) and their individual patches on our [security issues](https://github.com/chamilo/chamilo-lms/wiki/security-issues) page.

Help us keep your users secure by updating today!

Download Chamilo [from here](https://chamilo.org/download) or use your hosting provider's support to ask for an update.

**Comment by loicfreeman** [comment]
Bonjour 
j'ai installé chamilo 1.11.26 sur un server a la maison qui fonctionnait bien j'ai fait la mise a jour vers 1.11.28 comme il faut ,enfin je pense,mais quand je l'ai redémarré j'avais une page completement blanche.Après une journée de tentatives avortées j'ai fini par refaire une installe propre sur un autre disque dur. Le soucis c'est que je n'arrive pas a trouver comment importer l'ancienne config avec les cours sur la nouvelle version.3 jours que j'écume internet en quete de reponses meme CHATgpt ne met dit rien qui vaille .Pouvez-vous me donner une methode fiable car quand je  vais migrer vers la V2. ce sera surement le meme souci.
Merci d'avance

**Reply by ywarnier** [reply]
Bonjour @loicfreeman 
Assure-toi (si possible) que tu tournes bien sur PHP 7.4.
Après la mise à jour (en *écrasant* le répertoire de Chamilo 1.11.26), si tu as des problèmes de page blanche, essaie de vider le répertoire app/cache/twig. Il faut aussi veiller à ce que les répertoires app/cache/ et web/ soient autorisés en écriture par le serveur web (ça peut poser problème sinon).
L'error_log d'Apache permet généralement assez rapidement de détecter ce qui ne va pas.

La version 2.0 ne sera pas officiellement prévue pour une mise à jour mais bien de nouvelles installations. La procédure de mise à jour fonctionne mais n'a pas encore été suffisamment testée pour que nous puissions la garantir. La version 2.1 devrait offrir cette possibilité.

**Reply by loicfreeman** [reply]
Bonjour @ywarnier 
C'est bon j'ai reinstallé et tout fonctionne pour le moment .Merci à tous je continue à suivre ce projet qui me plait bien.
Cordialement Loic

**Reply by ywarnier** [reply]
Super, merci du feedback.

**Comment by Kaneda-1** [comment]
Bonjour @loicfreeman . Vous devez vérifier l'adresse suivante

https://11.chamilo.org/documentation/installation_guide.html

3.1 Upgrading from Chamilo 1.11.x (Minor version)
As this is only a minor version change from previous 1.11.* versions of Chamilo, the only thing you need to do is:
check that you haven't left any customised stylesheet or image (if you have, make sure you keep a copy on the side*)
download the Chamilo 1.11 installation package from the Chamilo download page
unzip the new files of Chamilo 1.11 over the files of the older version (or unzip the files in one folder and then copy the files from there to the older version's directory).
clean your app/cache/twig directory: delete all the contents *in* this directory (do NOT remove the directory itself, only its contents!). It will all be re-generated. You can also delete the contents of this directory through the "Cleanup of cache and temporary files " option in the "System" box of the Administration page.
edit app/config/configuration.php and change manually the $_configuration['system_version'] setting to the new version number
you're done! No other upgrade procedure is required

**Reply by loicfreeman** [reply]
Bonsoir @Kaneda-1 . C'est bien ce que j'ai fait pourtant . Comme j'avais fait une sauvegarde de mon disque avec la version1.11.26 ce que je voudrais savoir c'est quels fichiers ou dossiers remplacer sur ma nouvelle version 1.11.28 pour remettre les cours ainsi que les enseignants inscrits sans etre obligé de tout refaire a la main . Peut-etre avec la BDD je sais pas, si vous pouviez juste me dire dans quelle direction aller je me débrouillerai aprés.
Merci beaucoup

PS: En fait je viens de trouver le dossier app/courses et je pense que si je copie les dossiers ça doit le faire.

**Comment by Kaneda-1** [comment]
Bonjour @loicfreeman 

Vous n'avez pas mis à jour correctement Chamilo 1.11.26 vers 1.11.28.

Dans votre installation 1.11.26, vous devez écraser les fichiers/dossiers avec le contenu de la version 1.11.28 téléchargé depuis https://github.com/chamilo/chamilo-lms/releases/download/v1.11.28/chamilo-1.11.28 .fermeture éclair

Vous ne devez pas effectuer une nouvelle installation de la dernière version. Chamilo est déjà installé, vous n'avez pas besoin de l'installer à nouveau.

De cette façon, vous aurez la même base de données, les mêmes cours, professeurs, étudiants, etc.

`unzip the new files of Chamilo 1.11 over the files of the older version (or unzip the files in one folder and then copy the files from there to the older version's directory).
clean your app/cache/twig directory: delete all the contents in this directory (do NOT remove the directory itself, only its contents!). It will all be re-generated. You can also delete the contents of this directory through the "Cleanup of cache and temporary files " option in the "System" box of the Administration page.
edit app/config/configuration.php and change manually the $_configuration['system_version'] setting to the new version number
you're done! No other upgrade procedure is required`

**Reply by loicfreeman** [reply]
Merci @Kaneda-1 je réessayerais demain sur ma sauvegarde de faire comme vous m'avez dit et je vous tiens au courant.Est-ce que la procédure sera la meme pour la version 2 ou il faudra reinstaller?

**Comment by ywarnier** [comment]
:warning:  A little bug made its way in 1.11.28 without us catching it (see details [here](https://github.com/chamilo/chamilo-lms/issues/5887)). We should be releasing v1.11.30 to fix it. In the meantime, you can apply a one-line change as described [here](https://github.com/chamilo/chamilo-lms/commit/368dfaf932bd9084f60088b24495ed3aa8c6becc)

**Reply by andre0ani** [reply]
Bonjour,
Je viens de faire la mise à  jour de la 1.11.26 à la 1.11.28, en uploadant et écrasant les fichiers directement dans le répertoire  existant.
Tout semble bon, mais il m'indique toujours que j'ai la version 1.11.26. J'ai bien vidangé le cache, comme indiqué sur cette page : 
https://11.chamilo.org/documentation/installation_guide_fr_FR.html#3._Upgrade_from_a_previous_version_of
Mais je ne trouve pas le fichier où il faut modifier le numéro de version manuellement.

**Comment by NicoDucou** [comment]
Bonjour @andre0ani 

Le numéro de version est à modifier dans le fichier app/config/configuration.php il faut chercher la ligne avec :
`$_configuration['system_version'] = '1.11.26';`

Et le remplacer par : 
`$_configuration['system_version'] = '1.11.28';`

Cordialement

**Reply by andre0ani** [reply]
Merci beaucoup @NicoDucou 
C'est bon, fait.
Je ne sais pas comment je l'ai raté hier soir...
Il y a eu quelques soucis de droits avec les fichiers dans config, j'ai du les modifier.
Encore merci ;-)
